internal/provider/resource_kubernetes_cloud.go

```diff
@@ -0,0 +1,233 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the Apache License, Version 2.0, see LICENCE file for details.
+
+package provider
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/juju/terraform-provider-juju/internal/juju"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ resource.Resource = &kubernetesCloudResource{}
+var _ resource.ResourceWithConfigure = &kubernetesCloudResource{}
+var _ resource.ResourceWithImportState = &kubernetesCloudResource{}
+
+func NewKubernetesCloudResource() resource.Resource {
+	return &kubernetesCloudResource{}
+}
+
+type kubernetesCloudResource struct {
+	client *juju.Client
+
+	// subCtx is the context created with the new tflog subsystem for applications.
+	subCtx context.Context
+}
+
+type kubernetesCloudResourceModel struct {
+	CloudName         types.String `tfsdk:"name"`
+	KubernetesConfig  types.String `tfsdk:"kubernetesconfig"`
+	ParentCloudName   types.String `tfsdk:"parentcloudname"`
+	ParentCloudRegion types.String `tfsdk:"parentcloudregion"`
```

Comment:
todo: the tfsdk references should have underscore between the words. E.g. `kubernetes_config`. I missed this in the schema PR.